sky/clouds/aws.py

```diff
@@ -350,16 +349,21 @@ def _make(instance_list):
                     # Setting this to None as AWS doesn't separately bill /
                     # attach the accelerators.  Billed as part of the VM type.
                     accelerators=None,
+                    cpu=None,
                 )
                 resource_list.append(r)
             return resource_list
 
         # Currently, handle a filter on accelerators only.
         accelerators = resources.accelerators
         if accelerators is None:
-            # No requirements to filter, so just return a default VM type.
-            return (_make([AWS.get_default_instance_type()]),
-                    fuzzy_candidate_list)
+            # Return a default instance type with the given number of vCPUs.
+            default_instance_type = AWS.get_default_instance_type(
+                cpu=resources.cpu)
+            if default_instance_type is None:
+                return ([], fuzzy_candidate_list)
+            else:
+                return (_make([default_instance_type]), fuzzy_candidate_list)
```

Comment:
Was confused a bit. Maybe replace `fuzzy_candidate_list` on L341, 364, 366 with []?

Reply:
Yes, they all are []. I thought it's a style convention to clarify the meaning, and mimicked the previous code. Changed them to [].